observatory/src/components/NetworkGraphTab.vue

```diff
@@ -370,21 +371,74 @@ export default{
+            try {
+                this.user_query = document.getElementById('dropdown').value;
+            } catch (error) {
+                this.user_query = "http://dbpedia.org/resource/Spratton";
             }
-            //fetch('api/graph?query=SELECT DISTINCT ?s ?p ?o WHERE {<http://dbpedia.org/resource/Spratton> ?p ?o BIND(<http://dbpedia.org/resource/Spratton> AS ?s)} LIMIT ' + this.node_limit,{ //1st degree            
-            fetch("api/graph?query=SELECT DISTINCT ?s ?p ?o WHERE {<http://dbpedia.org/resource/Spratton> ?p1 ?s. ?s ?p ?o.} LIMIT " + this.node_limit,{ //2nd degree
-            //fetch("api/graph?query=SELECT DISTINCT ?s ?p ?o WHERE {<http://dbpedia.org/resource/Spratton> ?p1 ?s2. ?s2 ?p2 ?s. ?s ?p ?o.} LIMIT " + this.node_limit,{ //3rd degree
+            fetch("api/graph?query=SELECT DISTINCT ?s ?p ?o WHERE { BIND(<" + this.user_query + "> AS ?s) <" + this.user_query + "> ?p ?o.}" ,{
                 headers:{"Accept":"application/sparql-results+json"}
             })
                 .then(response=>response.json())
                 .then(response=>(this.post=response))
-                .then(response=>this.draw_graph(response));
-    }
+                .then(response=>this.draw_graph(response))
+                .catch((error) => {
+                    console.error('Error:', error);
+                });
+            fetch("api/graph?query=SELECT DISTINCT ?s ?p ?o WHERE { <" + this.user_query + "> ?p1 ?s. ?s ?p ?o.}" ,{
+            headers:{"Accept":"application/sparql-results+json"}
+            })
+                .then(response=>response.json())
+                .then(response=>(this.post=response))
+                .then(response=>this.draw_graph(response))
+                .catch((error) => {
+                    console.error('Error:', error);
+                });
+        },
+        resolveAfter3Seconds() {
+            return new Promise(resolve => {
+            setTimeout(() => {
+            resolve('resolved');
+            }, 3000);
+            });
+        },
+        async list_subjects(){
+            try {
+                for (let i = 0; i < this.subjects.results.bindings.length; i++) {
+                let subject = this.subjects.results.bindings[i].s.value;
+                this.subjects_list.push(subject);
+                }
+                this.fill_dropdown();
+            }
+            catch(error) {
+                const wait = await this.resolveAfter3Seconds();
+                this.list_subjects();
+            }
+        },
+        fetch_subjects(){ 
+            fetch("api/graph?query=SELECT DISTINCT ?s {?s ?p ?o}",{           
+            headers:{"Accept":"application/sparql-results+json"}
+            })
+                .then(response=>response.json())
+                .then(response=>(this.subjects = response))
+                .catch((error) => {
+                    console.error('Error:', error);
+                });
+            this.list_subjects();
```

Comment:
Shift 429 to be  a .then between 425 and 426. 
It'll mean it'll only do the list subjects method after fetching the subjects - meaning you won't need to do the resolve after 3 seconds bit, and means you can catch actual errors and do something more sensible with them in list_subjects, rather than just trying again in 3 seconds.